jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileToolkitCredentialsProviderFactory.kt

```diff
@@ -11,53 +11,88 @@ import software.aws.toolkits.core.credentials.ToolkitCredentialsProviderFactory
 import software.aws.toolkits.core.region.ToolkitRegionProvider
 import software.aws.toolkits.core.utils.tryOrNull
 import software.aws.toolkits.resources.message
-import java.nio.file.Path
 
 class ProfileToolkitCredentialsProviderFactory(
     private val sdkHttpClient: SdkHttpClient,
-    private val regionProvider: ToolkitRegionProvider,
-    private val credentialLocationOverride: Path? = null
-) : ToolkitCredentialsProviderFactory(TYPE) {
+    private val regionProvider: ToolkitRegionProvider
+) : ToolkitCredentialsProviderFactory<ProfileToolkitCredentialsProvider>(TYPE) {
+
+    private val profileWatcher = ProfileWatcher(::loadFromProfileFile)
+    private val profileHolder = ProfileHolder()
 
     init {
         loadFromProfileFile()
+
+        // TODO: If we remove a profile, we need to unselect it...do we need a listener then.......maybe at the manager level so it isnt profile specific..but that means passing manager into this class
+        // TODO: If .aws doesn't exist we never detect the changes...should we expose a way to reset the watcher and call it from CreateOrUpdateCredentialProfilesAction
```

Comment:
I think a reset makes the most sense since it's possible for a user to set their config/credential file locations to be `~/some/deep/non-existent/path` where none of the parent directories exist

Reply:
A simple approach could be to message the user that the configured location could not be found, and that the next attempt to watch it will take place the next time the IDE/plugin starts up.

Reply:
Ya, but that seems odd that we can edit/create them for them, but then they have to restart